### PR TITLE
Cluster: Add memberlist label configuration option

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
This PR makes it possible to set a label for the Memberlist cluster. This label is used to block any traffic that is not meant for the cluster.

For instance, in a situation where there are multiple alertmanager clusters, such as in K8s where IP addresses are shared and reused throughout the cluster, this label helps to prevent a wrong alertmanager cluster from receiving traffic meant for another cluster.

### CODE REVIEW

File: cmd/alertmanager/main.go [hunk 1]
```diff
@@ -226,6 +226,7 @@ func run() int {
 		peerReconnectTimeout   = kingpin.Flag("cluster.reconnect-timeout", "Length of time to attempt to reconnect to a lost peer.").Default(cluster.DefaultReconnectTimeout.String()).Duration()
 		tlsConfigFile          = kingpin.Flag("cluster.tls-config", "[EXPERIMENTAL] Path to config yaml file that can enable mutual TLS within the gossip protocol.").Default("").String()
 		allowInsecureAdvertise = kingpin.Flag("cluster.allow-insecure-public-advertise-address-discovery", "[EXPERIMENTAL] Allow alertmanager to discover and listen on a public IP address.").Bool()
+		label                  = kingpin.Flag("cluster.label", "A unique identifier for the cluster used to prevent cross-communication issues when sending gossip messages.").Default("").String()
```
Comment (gotjosh) [inline cmd/alertmanager/main.go:229]:
```suggestion
		label                  = kingpin.Flag("cluster.label", "The cluster label is an optional string to include on each packet and stream. It uniquely identifies the cluster and prevents cross-communication issues when sending gossip messages.").Default("").String()
```

 	The cluster label is an optional string to include in outbound packets and gossip streams. Other members in the memberlist cluster will discard any message whose label doesn't match the configured one, unless the 'cluster-label-verification-disabled' configuration option is set to true

**Review by gotjosh — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -372,6 +372,7 @@ be configured to communicate with each other. This is configured using the
 - `--cluster.probe-interval` value: interval between random node probes (default "1s")
 - `--cluster.reconnect-interval` value: interval between attempting to reconnect to lost peers (default "10s")
 - `--cluster.reconnect-timeout` value: length of time to attempt to reconnect to a lost peer (default: "6h0m0s")
+- `--cluster.label` value: the label is an optional string to include on each packet and stream. It uniquely identifies the cluster and prevents cross-communication issues when sending gossip messages(default:"")
```
Comment (JohnnyQQQQ) [inline README.md:375]:
```suggestion
- `--cluster.label` value: the label is an optional string to include on each packet and stream. It uniquely identifies the cluster and prevents cross-communication issues when sending gossip messages (default:"")
```

**Comment by gotjosh** [comment]
Thank you very much for your contribution!

**Pull request merged**